lib/rest-adapter.js

```diff
@@ -69,20 +70,31 @@ RestAdapter.createRestAdapter = function(remotes) {
  * Get the path for the given method.
  */
 
-RestAdapter.prototype.getRoutes = getRoutes;
-function getRoutes(obj) {
-  var routes = obj.http;
+RestAdapter.prototype.getRoutes = function(obj) {
+  return getRoutes(obj, this.options);
+};
 
+function getRoutes(obj, options) {
+  var routes = obj.http;
   if (routes && !Array.isArray(routes)) {
     routes = [routes];
   }
 
-  // overidden
+  // Options of obj (e.g. sharedClass) take precedence over options of adapter
+  const sharedClass = obj.sharedClass || obj;
```

Comment:
`obj` can be either a `SharedClass` instance (I think that's for constructors?) or a `SharedMethod` instance. In the second case we need to read the options from `SharedClass`.